autonomy_software/planning_pkg/src/target_generator.py

```diff
@@ -127,17 +129,38 @@ def callback_current_pose(self, pose_msg: PoseStamped):
             """ Get Next waypoint - First waypoint further than the lookahead distance """
             nextWaypointId = self.getNextWaypoint()
 
-            """ Get target point """
-            targetPoint = self.getTargetPoint(nextWaypointId)
+            # If next waypoint is the last one and we are not looping over, no need to
+            # try and find the target point. If we do, the target point will be behind us
+            # and this is a potential hazard. Instead, set target point to last waypoint.
+            # NOTE: This could theoretically be better if getNextWaypoint would publish something else when
+            # all the waypoints are within the lookahead
+            if nextWaypointId is None and self.loop_over_waypoints is False:
```

Comment:
Do you have the githooks enabled? I think black would suggest modifications here (maybe something like `if not nextWaypointId and not self.loop_over_waypoints:`). Not a big issue.

Reply:
I do yes. It actually suggested me that I think. I used to have something different. Should we change?

Reply:
No that's fine